test/system/src/test/java/io/pravega/test/system/ConsumptionBasedRetentionWithMultipleReaderGroupsTest.java

```diff
@@ -283,4 +286,140 @@ public void multipleSubscriberCBRTest() throws Exception {
+        // Check to make sure truncation happened at streamcut in the retention set
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 210.", true, () -> controller.getSegmentsAtTime(
+                        new StreamImpl(SCOPE_1, STREAM_1), 0L).join().values().stream().anyMatch(off -> off == 210),
+                5000, 3 * 60 * 1000L);
+
+        //Changing the readergroup from non-subscriber to subscriber again
+        readerGroup.resetReaderGroup(readerGroupConfig);
+        readerGroupRetentionType = controller.getReaderGroupConfig(SCOPE_1, READER_GROUP_3).join().getRetentionType();
+        assertEquals(ReaderGroupConfig.StreamDataRetention.MANUAL_RELEASE_AT_USER_STREAMCUT, readerGroupRetentionType);
+        assertEquals(1, controller.listSubscribers(SCOPE_1, STREAM_1).join().size());
+
+        // Recreates the reader
+        reader = clientFactory.createReader(READER_GROUP_3 + "-" + 1, READER_GROUP_3, new JavaSerializer<>(),
+                ReaderConfig.builder().build());
+
+        // fill stream with 3 events
+        for (int i = 1; i <= 3; i++) {
+            log.info("Writing event to {}/{}", SCOPE_1, STREAM_1);
+            writer.writeEvent(SIZE_30_EVENT).join();
+        }
+
+        //Read 3 events with reader from the segment offset
+        for (int i = 1; i <= 3; i++) {
+            read = reader.readNextEvent(READ_TIMEOUT);
+            assertEquals("data of size 30", read.getEvent());
+        }
+
+        log.info("{} generating 2nd stream-cuts for {}/{}", READER_GROUP_3, SCOPE_1, STREAM_1);
+        futureCuts = readerGroup.generateStreamCuts(streamCutExecutor);
+        // Wait for 5 seconds to force reader group state update. This will allow for the silent
+        // checkpoint event generated as part of generateStreamCuts to be picked and processed.
+        Futures.delayedFuture(Duration.ofSeconds(5), executor).join();
+        read = reader.readNextEvent(READ_TIMEOUT);
+        assertEquals("data of size 30", read.getEvent());
+        assertTrue("Stream-cut generation did not complete for reader group", Futures.await(futureCuts, 10000));
+        streamCuts = futureCuts.join();
+        log.info("{} updating its retention stream-cut to {}", READER_GROUP_3, streamCuts);
+        readerGroup.updateRetentionStreamCut(streamCuts);
+
+        // Retention set has two stream cut at 0/360, 0/450
+        // READER_GROUP_3 updated stream cut at 0/300, Subscriber lower bound is 0/300
+        // So truncating at stream cut 0/300 which leaves more data in the stream than min limit
+        // The timeout is set to 3 minutes a little longer than the retention period which is set to 2 minutes
+        // in order to confirm that the retention has taken place.
+        // Check to make sure truncation happened at SLB
+        AssertExtensions.assertEventuallyEquals("Truncation did not take place at offset 300.", true, () -> controller.getSegmentsAtTime(
```

Comment:
Same here: how long does it normally takes this assertion to complete?

Reply:
There is a property `controller.retention.frequency.minutes=2` which says retention is happening every 2 minutes.